### PR TITLE
log(engine): add Logger() interface to BaseMaster/BaseWorker/BaseJobMaster

### LINKED ISSUE #5803: engine:  BaseMaster/BaseWorker supports Logger() interface
### Is your feature request related to a problem?

No

### Describe the feature you'd like

BaseMaster/BaseWorker supports Logger() interface

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by maxshuang**
### What problem does this PR solve?

Issue Number: close #5803 

### What is changed and how it works?
1. add Logger() interface to BaseMaster/BaseWorker/BaseJobMaster
2. add L() for easy use

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: engine/lib/master.go [hunk 1]
```diff
@@ -108,6 +109,7 @@ type BaseMaster interface {
 	// MetaKVClient return user metastore kv client
 	MetaKVClient() metaclient.KVClient
 	MetricFactory() promutil.Factory
+	Logger() *zap.Logger
```
Comment (amyangfei) [inline engine/lib/master.go:112]: Where will the `Logger` of `BaseMaster` be used

Comment (amyangfei) [inline engine/lib/master.go:112]: Where will the `Logger` of `BaseMaster` be used

**Review by lance6716 — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**